pkg/auth/auth.go

```diff
@@ -270,6 +274,26 @@ func authConfigsToAuthFile(authConfigs map[string]types.DockerAuthConfig) (strin
 	return authFilePath, nil
 }
 
+// normalizeAuthFileKey takes a key, converts it to a host name and normalizes
+// the resulting registry.
+func normalizeAuthFileKey(key string) string {
+	stripped := strings.TrimPrefix(key, "http://")
+	stripped = strings.TrimPrefix(stripped, "https://")
+
+	// Assume that http(s) prefixed keys are not intended to be namespaced and
+	// normalize them to their base domain.
+	if stripped != key {
+		stripped = strings.SplitN(stripped, "/", 2)[0]
+	}
+
+	// Normalize legacy docker domains.
+	if stripped == "registry-1.docker.io" || stripped == "docker.io" {
+		return "index.docker.io"
```

Comment:
Doesn't that break keys with namespaces?

The normalization seems redundant  since `findAuthentication` already takes care of that. I think that stripping "http{s}://" is sufficient.

Reply:
Yeah I agree, changed to strip only the http[s]:// prefix